lib/ui/painting/image_decoding.cc

```diff
@@ -35,20 +35,20 @@ sk_sp<SkImage> DecodeImage(std::vector<uint8_t> buffer) {
   if (sk_data == nullptr)
     return nullptr;
 
-  std::unique_ptr<SkImageGenerator> generator(
-      SkImageGenerator::NewFromEncoded(sk_data.get()));
+  sk_sp<SkImage> image = SkImage::MakeFromEncoded(std::move(sk_data));
```

Comment:
This actually ends up calling the same code as before (and I've tested it with the makeTextureImage path disabled). Somehow, it works - although I agree that it shouldn't. Even the old code should have had this problem - SkImageGenerator::NewFromEncoded takes the raw SkData* and then turns it into an sk_sp. (SkCodecImageGenerator, line 21).